teammapper-frontend/mmp/src/map/handlers/draw.ts

```diff
@@ -55,13 +55,23 @@ export default class Draw {
      * Update the dom of the map with the (new) nodes.
      */
     public update() {
-        const nodes = this.map.nodes.getNodes(),
-            dom = {
-                nodes: this.map.dom.g.selectAll('.' + this.map.id + '_node').data(nodes),
-                branches: this.map.dom.g.selectAll('.' + this.map.id + '_branch').data(nodes.slice(1))
-            }
+        let nodes = this.map.nodes.getNodes().filter(x => !x.hidden)
+
+        const dom = {
+            nodes: this.map.dom.g.selectAll('.' + this.map.id + '_node').data(nodes, (d) => d.id),
+            branches: this.map.dom.g.selectAll('.' + this.map.id + '_branch').data(nodes.slice(1), (d) => d.id)
+        }
         let tapedTwice = false
 
+        dom.nodes.each((node: Node) => {
```

Comment:
wouldnt it be easier to set the hidden field directly when the button is pressed for all nodes, and not inside this render method?

Reply:
Not sure what you mean by this? There's practically no difference whether you actually set a field on the node itself or do it like this, as you'd still have to call both methods (`setHiddenChildrenIcon` and `removeHiddenChildrenIcon`) separately whenever you add/remove the icon to make it work reliably, at least in my testing.

You also can't avoid calling `dom.nodes.each` as `outer.each` (which is based on the `enter` method of D3.js) only seems to include nodes that were changed, ie those that were hidden, whereas we explicitly want the node that wasn't hidden.